test/jdk/java/awt/Dialog/DialogDisposeLeak.java

```diff
@@ -0,0 +1,98 @@
+/*
+ * Copyright (c) 1998, 2024, Oracle and/or its affiliates. All rights reserved.
+ * DO NOT ALTER OR REMOVE COPYRIGHT NOTICES OR THIS FILE HEADER.
+ *
+ * This code is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU General Public License version 2 only, as
+ * published by the Free Software Foundation.
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+
+import java.awt.Dialog;
+import java.awt.EventQueue;
+import java.awt.FlowLayout;
+import java.awt.Frame;
+import java.awt.Label;
+
+/*
+ * @test
+ * @bug 4193022
+ * @summary Test for bug(s): 4193022, disposing dialog leaks memory
+ * @key headful
+ * @run main/othervm -Xmx128m DialogDisposeLeak
+ */
+
+public class DialogDisposeLeak {
```

Comment:
It looks like this test was oversimplified, as I no longer see steps to reproduce the original issue.

https://bugs.openjdk.org/browse/JDK-4193022

> Container.trackMouseEnterExit() calls startListeningForOtherDrags() when the mouse pointer enters the container. It calls stopListeninForOtherDrags() when the mouse exits the container.
>
> If the container is a window (in our case a JDialog), and the dispose() method is called while the mouse pointer is within the window (user clicks on the 'ok' button, for example), then the window is disposed, but no mouse exit event occurs.
>
> This causes the window and everything it points to not to be collectable, as there is still the Container drag listener pointing at the window.

Reply:
i will revert this test to a manual test